docs/howto/craft-a-snap/example-cross-compiled-app.rst

```diff
@@ -0,0 +1,161 @@
+
+Overview
+--------
+
+There are a number of steps to enable cross-compilation for an autotools project:
+
+#. Specify the architectures used.
+#. Adjust the options passed to the ``configure`` script.
+#. Include packages to support cross-compilation.
+#. Handle linker warnings.
+
+Example project file for example-curl
+-------------------------------------
+
+The following code comprises the project file for a project that uses ``libcurl``. The
+complete code can be found in `this repository <https://github.com/mr-cal/hello-curl>`_.
+
+.. collapse:: hello-curl project
+
+  .. literalinclude:: ../code/craft-a-snap/example-cross-compiled-app-recipe.yaml
+      :caption: snapcraft.yaml
+      :language: yaml
+
+The ``platforms`` key
+-------------------------
+
+The :ref:`platforms <architectures-howto>` key is normally used to specify which
+architectures should be used when building a snap. For example, in the following case
+the build architecture is the same as the target architecture:
+
+.. code:: yaml
+
+    platforms:
+      - amd64
+
+It is also possible to specify different architectures for the build process and the
+target system. This enables cross-compilation to be managed by Snapcraft.
+
+Enable cross-compilation
+------------------------
+
+The ``platforms`` key can also be used to specify sets of architectures using the
+``build-on`` and ``build-for`` sets. In the simplest case, this can be used to
+cross-compile on one architecture for a single, different architecture:
+
+.. code:: yaml
+
+    platforms:
+      - build-on: amd64
+        build-for: armhf
+
+In this example, Snapcraft will only build the project on the ``amd64`` architecture,
+and it will only build it for the ``armhf`` architecture.
+
+The ``hello-curl`` project uses the ``platforms`` key to list many cross-compilation
+targets:
+
+.. literalinclude:: ../code/craft-a-snap/example-cross-compiled-app-recipe.yaml
+   :caption: platforms
+   :language: yaml
+   :start-at: platforms:
+   :end-before: lint:
+
+Adjust the autotools configuration
+----------------------------------
+
+When building for a particular architecture, Snapcraft will initialise the
+``CRAFT_ARCH_TRIPLET_BUILD_FOR`` environment variable in the build environment. This variable
+describes the platform and architecture that autotools uses to configure
+cross-compilation. For more information on environment variables, see `environment
+variables <https://snapcraft.io/docs/parts-environment-variables>`_.
+
+A project that uses the autotools plugin can adjust the options passed to the
+``configure`` script for a given part by using the ``autotools-configure-parameters``
+key that the plugin provides. In the example project, the ``hello-curl`` part is
+configured this way:
+
+.. literalinclude:: ../code/craft-a-snap/example-cross-compiled-app-recipe.yaml
+   :caption: autotools-configure-parameters
+   :language: yaml
+   :start-at: autotools-configure-parameters:
+
+Supply the dependencies
```

Comment:
> Supply

What do you think about "satisfy" instead? "Supply" makes me think we're doing the work of copying the packages manually.